tests/features/joinup_group/short_id.feature

```diff
@@ -0,0 +1,130 @@
+Feature:
+  As an owner of a group
+  In order to make the URL of my group easier to read
+  I need to be able to provide a short ID to my group.
+
+  Scenario: Short IDs are unique in collections.
+    Given the following collections:
+      | title                     | short ID | state     |
+      | Scientific Studies        | SSt-2020 | validated |
+      | Scientific Sustainability | SST-2020 | validated |
+    And owner:
+      | name                 | type    |
+      | Organisation example | Company |
+
+    When I am logged in as an authenticated
+    And I go to the propose collection form
+
+    # Assert the Short ID description.
+    Then I should see the following lines of text:
+      | Short URL is an abbreviated reference (It serves as a "nickname") for the collection, like an acronym. E.g. for a collection named "European Commission" it can be "EuCo". This value will be displayed in your browser's address bar. |
+      | This is not editable after the collection's initial publication.                                                                                                                                                                       |
+      | The value must be between 4 and 26 characters and can contain only alphanumeric characters and the dash(-) symbol.                                                                                                                     |
+
+    When I press "Add existing" at the "Owner" field
+    And I fill in the following:
+      | Title       | Structural Solar Traces    |
+      | Description | Structural Solar Traces    |
+      | Owner       | Organisation example       |
+      # Contact information data.
+      | Name        | Contact person             |
+      | E-mail      | contact_person@example.com |
+    And I select "HR" from "Policy domain"
+    And I fill in "Short ID" with "SST-2020"
+    And I press "Propose"
+    Then I should see the error message "Content with short id SST-2020 already exists. Please choose a different short id."
+
+    And I fill in "Short ID" with "Sst-2020"
+    And I press "Propose"
+    # The short id is case insensitive.
+    Then I should see the error message "Content with short id Sst-2020 already exists. Please choose a different short id."
+
+    And I fill in "Short ID" with "SsTr-2020"
+    And I press "Propose"
+    And I should see the heading "Structural Solar Traces"
+
+    When I go to the "Structural Solar Traces" collection edit form
+    Then the following fields should not be disabled "Short ID"
+
+    # Publish the collection.
+    When I am logged in as a moderator
+    And I go to the "Structural Solar Traces" collection edit form
+    And I press "Publish"
+    Then I should see the heading "Structural Solar Traces"
+
+    # Check that the "Short ID" field is not disabled for moderators.
+    When I click "Edit" in the "Entity actions" region
+    Then the following fields should not be disabled "Short ID"
```

Comment:
Can we also test that the moderator can change the value? Now we know that the field is not disabled, that's already a good thing, but maybe we get an error on save?

Reply:
But this is field API. What kind of error would you expect? There is already a uniqueness check in the first time that the entity is saved. There is no reason why the behaviour would change in the second one.